Building issue on Debian Jessie

I have installed OSRM before on Ubuntu and it has worked perfectly for me. But now I am facing issues in trying to install OSRM on Debian Jessie. After the "cmake ..", when I try to execute "make", I am getting the below error. 
 ```
[  0%] Built target CONTRACTOR
[ 24%] Built target ENGINE
[ 57%] Built target EXTRACTOR
[ 59%] Configuring revision fingerprint
-- Fingerprint unchanged, not regenerating
[ 59%] Built target FingerPrintConfigure
[ 75%] Built target SERVER
[ 78%] Built target STORAGE
[ 93%] Built target UTIL
Scanning dependencies of target osrm
[ 95%] Building CXX object CMakeFiles/osrm.dir/src/osrm/osrm.cpp.o
In file included from /data/osrm_core/include/engine/api/match_parameters.hpp:31:0,
                 from /data/osrm_core/src/osrm/osrm.cpp:2:
/data/osrm_core/include/engine/api/route_parameters.hpp: In instantiation of ‘osrm::engine::api::RouteParameters::RouteParameters(bool, bool, bool, osrm::engine::api::RouteParameters::GeometriesType, osrm::engine::api::RouteParameters::OverviewType, boost::optional<bool>, Args ...) [with Args = {}]’:
/data/osrm_core/include/engine/api/match_parameters.hpp:59:29:   required from here
/data/osrm_core/include/engine/api/route_parameters.hpp:82:47: warning: missing initializer for member ‘osrm::engine::api::BaseParameters::coordinates’ [-Wmissing-field-initializers]
           continue_straight{continue_straight_}
                                               ^
/data/osrm_core/include/engine/api/route_parameters.hpp:82:47: warning: missing initializer for member ‘osrm::engine::api::BaseParameters::hints’ [-Wmissing-field-initializers]
/data/osrm_core/include/engine/api/route_parameters.hpp:82:47: warning: missing initializer for member ‘osrm::engine::api::BaseParameters::radiuses’ [-Wmissing-field-initializers]
/data/osrm_core/include/engine/api/route_parameters.hpp:82:47: warning: missing initializer for member ‘osrm::engine::api::BaseParameters::bearings’ [-Wmissing-field-initializers]
Linking CXX static library libosrm.a
[ 95%] Built target osrm
[ 95%] Built target osrm_contract
Linking CXX executable osrm-contract
/tmp/ccp8OR7F.ltrans9.ltrans.o:ccp8OR7F.ltrans9.o:vtable for tbb::empty_task: error: undefined reference to 'tbb::empty_task::~empty_task()'
/tmp/ccp8OR7F.ltrans9.ltrans.o:ccp8OR7F.ltrans9.o:vtable for tbb::empty_task: error: undefined reference to 'tbb::empty_task::~empty_task()'
collect2: error: ld returned 1 exit status
CMakeFiles/osrm-contract.dir/build.make:109: recipe for target 'osrm-contract' failed
make[2]: *** [osrm-contract] Error 1
CMakeFiles/Makefile2:342: recipe for target 'CMakeFiles/osrm-contract.dir/all' failed
make[1]: *** [CMakeFiles/osrm-contract.dir/all] Error 2
Makefile:117: recipe for target 'all' failed
make: *** [all] Error 2

```

Thanks for reporting,

```
/tmp/ccp8OR7F.ltrans9.ltrans.o:ccp8OR7F.ltrans9.o:vtable for tbb::empty_task: error: undefined reference to 'tbb::empty_task::~empty_task()'
/tmp/ccp8OR7F.ltrans9.ltrans.o:ccp8OR7F.ltrans9.o:vtable for tbb::empty_task: error: undefined reference to 'tbb::empty_task::~empty_task()'
```

Can reproduce in a fresh Jessie container

    docker run -i -v $(pwd):/app -t debian:jessie /bin/bash

with packages from apt

    apt-get update && apt-get install --yes --no-install-recommends build-essential git cmake pkg-config libbz2-dev libstxxl-dev libxml2-dev libzip-dev libboost-all-dev liblua5.1-0-dev libluabind-dev libtbb-dev

Looking into this. My first guess is either LTO or our recent Gold linker modifications.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -126,8 +126,8 @@ if(ENABLE_GOLD_LINKER)
 
         # Issue 2785: check gold binutils version and don't use gc-sections for versions prior 2.25
         string(REGEX REPLACE ".*\\(GNU Binutils[^\\)0-9]+([0-9]+\\.[0-9]+)[^\\)]*\\).*" "\\1" GOLD_BINUTILS_VERSION "${LD_VERSION}")
-        if ("${GOLD_BINUTILS_VERSION}" VERSION_LESS "2.25")
-          message(STATUS "Disabling gc-sections on gold binutils < 2.25, see: https://sourceware.org/bugzilla/show_bug.cgi?id=17639")
+        if ("${GOLD_BINUTILS_VERSION}" VERSION_LESS "2.26")
+          message(STATUS "Disabling gc-sections on gold binutils < 2.26, see: https://sourceware.org/bugzilla/show_bug.cgi?id=17639")
           set(LD_AVOID_GC_SECTIONS TRUE)
         endif()
     else()

